source/simulator/solver_schemes.cc

```diff
@@ -459,8 +459,11 @@ namespace aspect
 
         assemble_newton_stokes_system = true;
 
-        solution.block(introspection.block_indices.pressure) = 0;
-        solution.block(introspection.block_indices.velocities) = 0;
+        if (nonlinear_iteration != 0)
+          {
+            solution.block(introspection.block_indices.pressure) = 0;
+            solution.block(introspection.block_indices.velocities) = 0;
```

Comment:
can you add a comment here that explains what you are doing? I suppose what you are doing is to set the iterate delta u to zero whenever you're not on the first Newton step, but leave it as is if you *are* on the first step?